R/Spectra-estimatePrecursorMz.R

```diff
@@ -0,0 +1,137 @@
+#' @title Estimating precursor m/z valus for DDA data
+#'
+#' @description
+#'
+#' MS data from Waters instruments are calibrated through the *Lock Mass*, but,
+#' while all m/z values of mass peaks in each spectrum will be calibrated by
+#' this method, the reported precursor m/z might not. The precursor m/z in the
+#' converted mzML file will have m/z values from quadrupole isolation windows
+#' instead of accurate m/z values. See also the
+#' [GNPS documentation](https://ccms-ucsd.github.io/GNPSDocumentation/fileconversion_waters/)
+#' for more information.
+#'
+#' The `estimatePrecursorMz` function estimates/adjusts the reported precursor
+#' m/z of a fragment spectrum using the following approach: in data dependent
+#' acquisition (DDA) mode, the MS instrument will select ions with the highest
+#' intensities in one MS scan for fragmentation. Thus, for each fragment
+#' spectrum, this method identifies the peak with the highest intensity and
+#' an m/z value similar to the fragment spectrum's reported precursor m/z
+#' (given parameters `tolerance` and `ppm`). This m/z value is then reported.
+#' Since the fragment spectrum's potential MS1 mass peak is selected based on
+#' its intensity, this method should **only be used for DDA data**.
+#'
+#' @note
+#'
+#' This approach should only be applied if fragment spectra are recorded
+#' through  data dependent acquisition (DDA), since only with this data it
```

Comment:
there's an extra space between "through" and "data".

Reply:
Also I'm a bit confused by the sentence, maybe rephrasing would help: 

"This approach is applicable only when fragment spectra are obtained through data-dependent acquisition (DDA), as it assumes that the peak with the highest intensity within the given m/z window corresponds to the precursor ion."

Reply:
Oh yes, great, thanks! that's much better. I'll fix.